packages/core/src/routes/admin-user.ts

```diff
@@ -130,6 +131,20 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
         await clearUserCustomDataById(userId);
       }
 
+      // Temp solution to validate the existence of input roleNames
+      if (body.roleNames) {
+        const { roleNames } = body;
+        const roles = await findRolesByRoleNames(roleNames);
+
+        if (roles.length !== roleNames.length) {
+          const resourcesNotFound = roleNames.filter(
+            (roleName) => !roles.some(({ name }) => roleName === name)
+          );
+          // TODO: Should be cached by the error handler and return request error
+          throw new InvalidInputError(`role names (${resourcesNotFound.join(',')}) are not valid`);
+        }
+      }
```

Comment:
nit: better compute the intersection of `roleNames` and `roles.name`s

Reply:
This is old code.

Reply:
There is a bug.
If lengths are the same and the content is different, it will not throw.

Reply:
YES, but this is a "Temp solution" (see the first line of comment)